middleware-jackson/src/main/java/com/nytimes/android/external/store3/middleware/jackson/JacksonReaderParser.kt

```diff
@@ -0,0 +1,43 @@
+package com.nytimes.android.external.store3.middleware.jackson
+
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.databind.JavaType
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.nytimes.android.external.store3.base.Parser
+import com.nytimes.android.external.store3.util.ParserException
+
+import java.io.IOException
+import java.io.Reader
+import java.lang.reflect.Type
+import javax.inject.Inject
+
+import io.reactivex.annotations.NonNull
+
+class JacksonReaderParser<Parsed> : Parser<Reader, Parsed> {
+
+    private val objectMapper: ObjectMapper
+    private val parsedType: JavaType
+
+    constructor(jsonFactory: JsonFactory, type: Type) {
+        objectMapper = ObjectMapper(jsonFactory).registerModule(KotlinModule())
+        parsedType = objectMapper.constructType(type)
+    }
+
+    @Inject
+    constructor(objectMapper: ObjectMapper, type: Type) {
+        this.objectMapper = objectMapper
+        parsedType = objectMapper.constructType(type)
+    }
+
+    @Throws(ParserException::class)
+    override fun apply(@NonNull reader: Reader): Parsed {
+        try {
```

Comment:
`return` can be lift here as well! Not sure, but I thought that the IDE offers a Refactoring Intention for it as well!